pkg/controller/sparkapplication/controller.go

```diff
@@ -1001,3 +1009,17 @@ func (c *Controller) hasApplicationExpired(app *v1beta2.SparkApplication) bool {
 
 	return false
 }
+
+// Clean up resources such as batch scheduler config if applicable.
+func (c *Controller) cleanUpOnTermination(oldApp, newApp *v1beta2.SparkApplication) error {
+	if needScheduling, scheduler := c.shouldDoBatchScheduling(newApp); needScheduling {
+		// If new app state is completed or failed, no more app is running,
```

Comment:
Comment needs update.

Reply:
updated it
`batch schduler is cleaned up only when state is changed to completion state`